src/schematools/utils.py

```diff
@@ -256,6 +256,23 @@ def dataset_schema_from_id_and_schemas_path(
     return dataset_schema
 
 
+def dataset_schemas_from_schemas_path(schemas_path: Union[Path, str]) -> Dict[str, types.ST]:
+    """Read all datasets from the schemas_path.
+
+    Args:
+        schemas_path: Path to the filesystem location with the dataset schemas.
+    """
+    schema_lookup: Dict[str, types.ST] = {}
+    for root, _, files in os.walk(schemas_path):
+        if "dataset.json" in set(files):
```

Comment:
```suggestion
        if "dataset.json" in files:
```

Er is niets efficients aan om een bestaande list eerst compleet om te zetten naar een set (en entries te hashen) zodat je "efficient" een IN operator kan gebruiken. Gewoon direct de lijst doorzoeken.

Reply:
Yes, idd. Alleen nuttig in bijv. een for-loop.